dask_jobqueue/core.py

```diff
@@ -414,18 +412,33 @@ def __init__(
         **kwargs
     ):
         self.status = "created"
+
+        default_job_cls = getattr(type(self), "job_cls", None)
+        self.job_cls = default_job_cls
         if job_cls is not None:
             self.job_cls = job_cls
 
+        # TODO better error
         if self.job_cls is None:
             raise ValueError(
-                "You must provide a Job type like PBSJob, SLURMJob, "
-                "or SGEJob with the job_cls= argument."
+                "you need to specify a Job type. "
+                "two cases:\n"
+                "- you are inheriting from JobQueueCluster (most likely): you need to have a class variable"
+                " in your JobQueueCluster-derived class\n"
+                "- you are using JobQueueCluster directly (less likely, only useful for tests), "
+                "pass a  job_cls= parameter."
             )
 
-        if config_name:
-            if interface is None:
-                interface = dask.config.get("jobqueue.%s.interface" % config_name)
+        default_config_name = getattr(self.job_cls, "config_name", None)
```

Comment:
So the key is here, all the informaton is in the Job class, not in the JobQueueCuster one.